File: counting_sort.cpp
#include<iostream>
using namespace std;
void count_sort_function(int arr1[], int total_size_of_arr1, int total_range, int low_range)
{
	int x, y = 0, count_arr[total_range] = {0};

	for(x=0; x<total_size_of_arr1; x++)
		count_arr[arr1[x]-low_range]++;

	x=0;

	while(x < total_range)
	{
		l1:
		arr1[y] = low_range+x;
		y++;
		count_arr[x]--;


		if(count_arr[x] > 0)
		goto l1;

		x++;
	}
}

int main()
{
    int total_size_of_arr1, x, total_range, high_range, low_range;
	cout<<"\nEnter the total_size_of_arr1 : ";
	cin>>total_size_of_arr1;

	cout<<"\nEnter the low_range and high_range :";
	cin>>low_range>>high_range;

	int arr1[total_size_of_arr1];
    for(x=0;x<total_size_of_arr1;x++)
	{
		cout<<"Enter Array element:  "<<x+1<<": ";
		cin>>arr1[x];
	}
	total_range = high_range-low_range+1;

	count_sort_function(arr1, total_size_of_arr1, total_range,low_range);

	cout<<"\nSorted Elements: ";
	for (int x = 0; x < total_size_of_arr1; x++)
        cout<<arr1[x]<<" ";

	return 0;
}

File: counting_sort.py
def counting_sort_function(Arr1):
    total_size_of_Arr1 = len(Arr1)
    temp1 = [0] * total_size_of_Arr1

    temp2 = [0] * 10
    
    for x in range(0, total_size_of_Arr1):
        temp2[Arr1[x]] += 1
        
    for x in range(1, 10):
        temp2[x] += temp2[x - 1]

    x = total_size_of_Arr1 - 1
    while x >= 0:
        temp1[temp2[Arr1[x]] - 1] = Arr1[x]
        temp2[Arr1[x]] -= 1
        x -= 1

    for x in range(0, total_size_of_Arr1):
        Arr1[x] = temp1[x]


Arr1 = [5,4, 7,2,5 , 9, 0, 3, 1,7,3]
counting_sort_function(Arr1)
print("Sorted Elements: ")
print(Arr1)
